### PR TITLE
fix(snownet,relay): re-use channels to peers in cooldown period

### PR DESCRIPTION
**Pull request by thomaseizinger**
For efficiency reasons, TURN's data channels don't have any authentication or integrity metadata. Instead, the operate using a short 2-byte channel number to identify the target peer of the data.

To avoid abuse, channel bindings are at most valid for 10 minutes before they need to be refreshed. In case they expire, there is a 5 minute cooldown period, before the same channel number can be bound to a different peer and before the same peer can be bound to a different channel.

We had a similar issue in the past (#5613) where channels got rebound early. Whilst that was fixed and is no longer happening, a case that we didn't consider is what happens if we want to bind a channel to a peer that still has a channel bound but is currently cooling down (i.e. in the 5 minute period after its expiry).

In that case, `snownet` would wrongly assume that there is no channel to this peer and try to bind a new one. That would get rejected by the relay with a bad request.

To fix this, we simply need to check whether we still have a channel to this peer and if yes, return the same channel number. On the relay, we need to ensure that we consider a channel as `bound` again when it is being refreshed.

We ensure that this doesn't regress in two ways:

- We add a unit-test for the `ChannelBindings` struct
- We modify the `Idle` transition to idle for 6 instead of 5 minutes. This ensures that a combination of 2 idle transitions puts the channel bindings into the 10-15 minute time window where rebinding the peer to a different channel fails.

Related: #6265.

### CODE REVIEW

File: website/src/components/Changelog/GUI.tsx [hunk 1]
```diff
@@ -13,12 +13,17 @@ export default function GUI({ title }: { title: string }) {
...
         <ul className="list-disc space-y-2 pl-4 mb-4">
           <ChangeItem pull="6226">
             Fixes a bug where clearing the log files would delete the current files, preventing logs from being written.
           </ChangeItem>
         </ul>
+        <ul className="list-disc space-y-2 pl-4 mb-4">
+          <ChangeItem pull="6276">
+            Fixes a bug where relayed connections failed to establish after an idle period.
+          </ChangeItem>
+        </ul>
```
Comment (jamilbk) [inline website/src/components/Changelog/GUI.tsx]:
```suggestion
          <ChangeItem pull="6276">
            Fixes a bug where relayed connections failed to establish after an idle period.
          </ChangeItem>
```

Comment (jamilbk) [reply website/src/components/Changelog/GUI.tsx]: repeated the `<ul>` here I think by accident

File: website/src/components/Changelog/Android.tsx [hunk 2]
```diff
@@ -15,10 +15,17 @@ export default function Android() {
...
+        <ul className="list-disc space-y-2 pl-4 mb-4">
           <ChangeItem pull="6181">
             Improves reliability of DNS resolution of non-resources.
           </ChangeItem>
         </ul>
+        <ul className="list-disc space-y-2 pl-4 mb-4">
+          <ChangeItem pull="6276">
+            Fixes a bug where relayed connections failed to establish after an idle period.
+          </ChangeItem>
+        </ul>
```
Comment (jamilbk) [inline website/src/components/Changelog/Android.tsx]:
```suggestion
          <ChangeItem pull="6181">
            Improves reliability of DNS resolution of non-resources.
          </ChangeItem>
          <ChangeItem pull="6276">
            Fixes a bug where relayed connections failed to establish after an idle period.
          </ChangeItem>
        </ul>
```

`ChangeItem`s are `<li>`s

**Review by jamilbk — CHANGES_REQUESTED** [review]

File: website/src/components/Changelog/GUI.tsx (hunk 1 above)
Comment (thomaseizinger) [reply website/src/components/Changelog/GUI.tsx]: Ah yes, my HTML-fu is a bit dated.

**Review by jamilbk — APPROVED** [review]

**Comment by jamilbk** [comment]
@thomaseizinger Ready to merge?

**Comment by conectado** [comment]
To make sure I understand this, the 5 minutes cooldown was the problem? So the binding has expired but we could only use the same channel?

**Comment by jamilbk** [comment]
> To make sure I understand this, the 5 minutes cooldown was the problem? So the binding has expired but we could only use the same channel?

Yes, the client would try to rebind the same channel, but since the Relay considered it cooling down, would reject.

**Comment by thomaseizinger** [comment]
> > To make sure I understand this, the 5 minutes cooldown was the problem? So the binding has expired but we could only use the same channel?
> 
> Yes, the client would try to rebind the same channel, but since the Relay considered it cooling down, would reject.

For the sake of correctness: The client would try to bind a **new** channel for a peer that still had a channel in cooldown phase. That is not allowed because it would make it ambigous, which channel to forward data to for incoming traffic from that peer.

The fix is to re-activate the already existing channel.

**Pull request merged**